test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,840 @@ TEST_F(SearchFilter, ExcludeRamp) {
+      close_single_edge(reader, tile, index, current, "AB", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    auto result = gurka::route(closure_map, {"1", "2"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"HIC", "CD"});
+    gurka::assert::raw::expect_path(result, {"HIC", "CD", "DE"});
+
+    // Specify search filter to disable exclude_closures at departure
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("2").lat()) %
+         std::to_string(closure_map.nodes.at("2").lng()) % costing % costing % date_type)
+            .str();
+    result = gurka::route(closure_map, req_disable_exclude_closures, reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE"});
+  }
+}
+
+/*
+ *  Tests search_filter.exclude_closures at destination
+ */
+TEST_P(ExcludeClosuresOnWaypoints, ExcludeClosuresAtDestination) {
+  std::string costing = std::get<0>(GetParam());
+  std::string date_type = std::get<1>(GetParam());
+  std::string costing_speed_type =
+      (boost::format("/costing_options/%s/speed_types/0") % costing).str();
+
+  // None of the edges are closed
+  {
+    auto result = gurka::route(closure_map, {"1", "3"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EJ", "JK"});
+  }
+  // DE edge is closed in both directions. Route should avoid DE with
+  // exclude_closures set to true (default) & and use it otherwise
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "JK", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    // TODO: 1->2 fails to get route without ignoring closure at 2. why?
```

Comment:
I found it odd that if I close edge `DE` and route to `2`, its unable to find the route. However the same does not happen when I close `JK` & route to `3`. I haven't found a good reason for this difference in behavior yet.

CC: @kevinkreiser @danpat @danpaz

Reply:
@mandeepsandhu I wonder if it's node snapping since DE is shorter than JK. Can you try increasing the gridsize from 10 to 100 and see if the same happens?

Reply:
Changing the gridsize to 100 does stop the no route exception. When you say "node snapping", do you mean its snapping to `E`? If so, shouldn't it still route upto `D`?

Reply:
Actually I also had to increase DE's length, otherwise its still unable to find the route